src/adapter/src/catalog/state.rs

```diff
@@ -2105,3 +2107,95 @@ impl ConnectionResolver for CatalogState {
         }
     }
 }
+
+impl ExprHumanizer for CatalogState {
+    fn humanize_id(&self, id: GlobalId) -> Option<String> {
+        self.entry_by_id
+            .get(&id)
+            .map(|entry| entry.name())
+            .map(|name| self.resolve_full_name(name, None).to_string())
+    }
```

Comment:
Sadly this is not 100% correct. For example

```rust
impl ExprHumanizer for CatalogState {
    fn humanize_id(&self, id: GlobalId) -> Option<String> {
        self.entry_by_id
            .get(&id)
            .map(|entry| entry.name())
            .map(|name| self.resolve_full_name(name, None).to_string())
    }
    // ...
}
```
is different than
```rust
impl ExprHumanizer for ConnCatalog<'_> {
    fn humanize_id(&self, id: GlobalId) -> Option<String> {
        self.state
            .entry_by_id
            .get(&id)
            .map(|entry| entry.name())
            .map(|name| self.resolve_full_name(name).to_string())
    }
    // ...
}
```

Because in the `ConnCatalog` variant the `self.resolve_full_name(name)` call using the current connection, whereas `self.resolve_full_name(name, None)` will use the `SYSTEM_CONN_ID`.

I'm not sure if this changes anything anywhere of substance for us at the moment, but I'm willing to give this a try and report what breaks. In general I would like to remove `ExprHumanizer for CatalogState` in favor of `ExprHumanizer for CatalogState` everywhere.

Reply:
If you are going to remove the old code, then the duplication is ok to have temporarily.

Reply:
Leaving the comment open so I mark `ExprHumanizer for ConnCatalog` as deprecated.